Sources/SourceKitLSP/TestDiscovery.swift

```diff
@@ -476,6 +477,42 @@ fileprivate extension Array<AnnotatedTestItem> {
     }
     return result
   }
+
+  func prefixTestsWithModuleName(workspace: Workspace) async -> Self {
+    return await self.asyncMap({
+      // If the module name can't be determined we return the test item without a prefixed id.
+      guard let moduleName = await self.moduleName(from: workspace, for: $0.testItem.location.uri) else {
+        return $0
+      }
+      var newTest = $0.testItem
+      newTest.id = "\(moduleName).\(newTest.id)"
+      newTest.children = await prefixTestsWithModuleName(workspace: workspace, newTest.children)
+      return AnnotatedTestItem(testItem: newTest, isExtension: $0.isExtension)
+    })
+  }
+
+  private func prefixTestsWithModuleName(workspace: Workspace, _ tests: [TestItem]) async -> [TestItem] {
```

Comment:
Should this be an extension on `Array<TestItem>`? Also: Could we share the implementation logic with `AnnotatedTestItem`?